### PR TITLE
add router to the app root

### PR DESCRIPTION
**Pull request by freben**
Follow-up of #22197

One question for all of these is naming - it's `createAppRootWrapper` and `createAppRootElement`, but `createRouter` (no App or AppRoot). Suggestions are welcome of course

### CODE REVIEW

File: packages/frontend-plugin-api/package.json [hunk 1]
```diff
@@ -22,6 +22,21 @@
     "prepack": "backstage-cli package prepack",
     "postpack": "backstage-cli package postpack"
   },
+  "dependencies": {
```
Comment (freben) [inline packages/frontend-plugin-api/package.json:25]: just putting these in the standard order :)

File: packages/frontend-test-utils/src/app/createExtensionTester.tsx [hunk 2]
```diff
@@ -302,7 +178,12 @@ export class ExtensionTester {
           extensions: [
             ...this.#extensions.map(extension => extension.definition),
             TestAppNavExtension,
-            TestAppRootExtension,
+            createRouterExtension({
+              namespace: 'test',
+              Component: ({ children }) => (
+                <MemoryRouter>{children}</MemoryRouter>
+              ),
+            }),
```
Comment (Rugvip) [inline packages/frontend-test-utils/src/app/createExtensionTester.tsx:186]: Nice 👌

File: packages/frontend-plugin-api/src/extensions/createRouterExtension.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import { Expand } from '../types';
+
+/**
+ * Creates an extension that replaces the router implementation at the app root.
+ * This is useful to be able to for example replace the BrowserRouter with a
+ * MemoryRouter in tests, or to add additional props to a BrowserRouter.
+ *
+ * @public
+ */
+export function createRouterExtension<
```
Comment (Rugvip) [inline packages/frontend-plugin-api/src/extensions/createRouterExtension.tsx:35]: It does feel very overkill this 🤔. I do think we can ship it for now though, and considering options to this pattern is already in our backlog.

**Review by Rugvip — APPROVED** [review]
Nice 👌

File: packages/frontend-test-utils/src/app/createExtensionTester.tsx (hunk 2 above)
Comment (camilaibs) [reply packages/frontend-test-utils/src/app/createExtensionTester.tsx:186]: 👏🏻

**Pull request merged**